devito/ir/iet/nodes.py

```diff
@@ -419,6 +486,29 @@ def children(self):
         """Return the traversable children."""
         return (self.nodes,)
 
+    @property
+    def functions(self):
```

Comment:
I'm adding this comment here, but it applies to lots of places. I think all these methods, ``functions, free_symbols, ...`` should return tuples, not lists

Reply:
In fact, after skimming through the changes, it seems like there are some inconsistencies ?

Reply:
This probably needs to be addressed

Reply:
yes